src/store/modules/permissions.js

```diff
@@ -1,55 +1,94 @@
+import Vue from 'vue';
 import { aettosToAe } from '../../popup/utils/helper';
 
+export const hostConfig = (addresses, isDefault) => ({
+  addresses,
+  address: isDefault,
+  messageSign: isDefault,
+  transactionSignLimit: isDefault ? 50 : 0,
+  transactionSignLimitLeft: isDefault ? 50 : 0,
+  transactionSignFirstAskedOn: null,
+});
+
+export class App {
+  constructor(host) {
+    this.host = host;
+  }
+}
+
 export default {
   namespaced: true,
 
   state: {
-    address: true,
-    messageSign: true,
-    transactionSignLimit: 0,
-    transactionSignLimitLeft: 0,
-    transactionSignFirstAskedOn: null,
+    ...['superhero.com', 'localhost'].reduce(
+      (st, host) => ({ ...st, [host]: hostConfig([], true) }),
+      {},
+    ),
   },
```

Comment:
probably it may be more redable using a new `Object.fromEntries` 🎉
```js
  state: Object.fromEntries(
    ['superhero.com', 'localhost'].map(domain => [domain, hostConfig([], true)])
  ),
```
so, no more `reduce` for collecting objects